internal/pkg/agent/cmd/container.go

```diff
@@ -1031,7 +1045,21 @@ func shouldFleetEnroll(setupCfg setupConfig) (bool, error) {
 		return true, nil
 	}
 
-	ctx := context.Background()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Check if enrolling with a specifically defined Elastic Agent ID.
+	// If the ID's don't match then it needs to enroll.
+	if setupCfg.Fleet.ID != "" {
+		ai, err := getAgentInfo(ctx, false)
```

Comment:
`getAgentInfo` seems to be loading the config exactly like we do below. Since you need only the ID can't we just use  `storedConfig.Settings.ID` right after the `storedConfig, err := configuration.NewFromConfig(cfg)`  below?

Reply:
Duh, I was so used to using the `AgentInfo` flow all over the code that I didn't even look at how it was working under the hood and see that it was loading the same encrypted store.

I have updated it to just check the results from the encrypted store.